src/pages/[lang]/punti-di-ritiro-2a89b635-66f8-458a-a59b-8fb4146cd9d7.tsx

```diff
@@ -103,13 +103,28 @@ const RitiroPage: NextPage = () => {
   };
 
   const handleSearchClick = () => {
-    const operators = initialRaddOperators.filter((operator) =>
-      operator.city
-        ? operator.city.toLowerCase().replace(/[^a-zA-Z]/g, "") ===
-            searchValue.toLowerCase().replace(/[^a-zA-Z]/g, "") ||
-          operator.cap === searchValue
-        : ""
-    );
+    const normalizedSearchValue = searchValue
+      .toLowerCase()
+      .replace(/[^a-zA-Z]/g, "");
+
+    const operators = initialRaddOperators.filter((operator) => {
+      if (!operator.city) {
```

Comment:
In the old logic if operator.city was undefined, we skip the filter for that operator. Instead of skipping validation or filtering only for cap, i should add a check like this

```
const normalizedCity = operator.city ? operator.city
        .toLowerCase()
        .replace(/[^a-zA-Z]/g, "") : '';
```

on city and on province. After that i should remove the if on operator.city